src/transformers/models/splinter/tokenization_splinter.py

```diff
@@ -0,0 +1,164 @@
+# coding=utf-8
+# Copyright Ori Ram, Yuval Kirstain, Jonathan Berant, Amir Globerson, Omer Levy and The HuggingFace Inc. team.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for Splinter."""
+from typing import List, Optional
+
+from ...utils import logging
+from ..bert.tokenization_bert import BertTokenizer
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.txt"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "splinter-base": "https://huggingface.co/splinter-base/resolve/main/vocab.txt",
+        "splinter-large": "https://huggingface.co/splinter-large/resolve/main/vocab.txt",
+    }
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "splinter-base": 512,
+    "splinter-large": 512,
+}
+
+
+PRETRAINED_INIT_CONFIGURATION = {
+    "splinter-base": {"do_lower_case": False},
+    "splinter-large": {"do_lower_case": False},
+}
+
+
+class SplinterTokenizer(BertTokenizer):
```

Comment:
Let's not abstract from a different tokenizer here. It's better to just copy-paste the code and make use of `#Copied from`

Reply:
@patrickvonplaten We do have a bit of a different tokenization (mainly in terms of `build_inputs_with_special_tokens`)

Reply:
Sorry, I wasn't super clear here I think. We are trying to remove as many abstractions as possible which in this case would mean that we don't abstract from the `BertTokentizer` class. Instead it would be better to just have: 

```python
class SplinterTokenizer:
```

and just copy paste all the required functionality from the BertTokenizer